packages/ckb/src/spore/cluster.ts

```diff
@@ -0,0 +1,96 @@
+import { RgbppCkbVirtualTx } from '../types/rgbpp';
+import { packRawClusterData } from '@spore-sdk/core';
+import { append0x, calculateTransactionFee } from '../utils';
+import { buildPreLockArgs, calculateCommitment, genRgbppLockScript } from '../utils/rgbpp';
+import { CreateClusterCkbVirtualTxParams, Hex, SporeVirtualTxResult } from '../types';
+import {
+  RGBPP_TX_WITNESS_MAX_SIZE,
+  RGBPP_WITNESS_PLACEHOLDER,
+  getClusterTypeDep,
+  getClusterTypeScript,
+  getRgbppLockConfigDep,
+  getRgbppLockDep,
+  getRgbppLockScript,
+} from '../constants';
+import { generateClusterCreateCoBuild, generateClusterId } from '../utils/spore';
+import { NoRgbppLiveCellError } from '../error';
+import { bytesToHex, getTransactionSize } from '@nervosnetwork/ckb-sdk-utils';
+
+/**
+ * Generate the virtual ckb transaction for creating cluster
+ * @param collector The collector that collects CKB live cells and transactions
+ * @param rgbppLockArgs The rgbpp assets cell lock script args whose data structure is: out_index | bitcoin_tx_id
+ * @param clusterData The cluster's data, including name and description.
+ * @param witnessLockPlaceholderSize The WitnessArgs.lock placeholder bytes array size and the default value is 3000 (It can make most scenarios work properly)
+ * @param ckbFeeRate The CKB transaction fee rate, default value is 1100
+ */
+export const genCreateClusterCkbVirtualTx = async ({
+  collector,
+  rgbppLockArgs,
+  clusterData,
+  isMainnet,
+  witnessLockPlaceholderSize,
+  ckbFeeRate,
+}: CreateClusterCkbVirtualTxParams): Promise<SporeVirtualTxResult> => {
+  const rgbppLock = {
+    ...getRgbppLockScript(isMainnet),
+    args: append0x(rgbppLockArgs),
+  };
+  const rgbppCells = await collector.getCells({ lock: rgbppLock });
+  if (!rgbppCells || rgbppCells.length === 0) {
+    throw new NoRgbppLiveCellError('No rgbpp cells found with the rgbpp lock args');
+  }
+  const rgbppCell = rgbppCells[0];
```

Comment:
Only unlocking rgbppCells[0] might not be enough.
related issue:
- https://github.com/ckb-cell/rgbpp-sdk/issues/66
- https://github.com/ckb-cell/rgbpp-sdk/issues/23

I think we could refactor it in another PR, what do you think? @duanyytop

Reply:
Agree.

My initial design was to simplify the creation of clusters and spores and lower the threshold for use by application teams. A universal design is indeed needed.